1.fee_bumping/fee_bumping.md

```diff
@@ -153,25 +153,217 @@ Bitcoin blockchain, such as exchanges or custodians:
+> 
+> It's easier said than implemented.  There would be a fair amount of work[...]
+
+There are two main areas of difficulty that Satoshi is talking about: wallet
+design and UX, and node and miner policy.
+
+The wallet design and UX questions (how should a wallet track and display
+conflicting transactions? How should it present a user interface for bumping
+fee?) can be resolved by individual wallets and applications.
+
+The node and miner policy difficulties require some kind of shared
+understanding of how other nodes and miners on the network will behave. There
+are two major difficulties to consider:
+
+1. **DoS** - how do we prevent malicious actors from consuming node resources
+   by repeatedly broadcasting replacement transactions for zero cost?
+2. **Incentive alignment** - how do we make sure that replacement transactions
+   are incentive-compatible and miners want to mine them?
+
+Satoshi's first version of transaction replacement failed on both of these
+counts. The `nSequence` number could be incremented millions of times without
+changing the fee, forcing each node in the network to re-validate all of the
+transactions many times at zero cost to the attacker. The replacement logic also
+didn't consider the transaction fee at all, and would replace an old
+transaction with a new transaction even if the replacement had a lower fee. If
+a miner was running this software, the replacement behaviour would be costing
+him money since he'd be mining transactions with lower fees.
+
+For those reasons, Satoshi [disabled transaction replacement][disable replacement]
+in 2010. When asked later why he'd disabled that code, his response was:
+
+> Just to reduce surface area.  It wouldn't help with increasing tx fee. A tx
+> starts being valid at nLockTime.  It wouldn't work to have a tx that stops
+> being valid at a certain time; once a tx ever becomes valid, it must stay
+> valid permanently.
+
+Replacing a transaction does not invalidate the old transaction, so unless
+there is an incentive for the miner to use the new transaction, then just
+replacing an old transaction with one that has higher `nSequence` numbers
+is not incentive-compatible.
+
+## Replace-by-Fee
+
+Replace-by-fee is an effort to make transactions replaceable whilst avoiding
+the problems of exposing nodes to DoS vulnerabilities and misaligned
+incentives.  This is achieved by only allowing replacement if the replacement
+transaction has a higher fee than the replaced transaction. This prevents DoS
+attacks since every replacement transaction costs the attacker additional
+money, and ensures incentive alignment since the replacement transaction is
+financially advantageous to the miner.
+
+There are several [variants of replace-by-fee][rbf variants]. Almost all nodes
+on the Bitcoin are thought to implement 'opt-in RBF' as defined in [BIP 125][].
+
+Bitcoin Core may implement 'delayed RBF' at some point in the future. There
```

Comment:
Style nitpick: you use *emphasis* for some terms and 'quotes' for others.  This should probably be made consistent.

Reply:
I stumbled here a bit in the thread of your writing. You set me up to learn about opt-in RBF, and then there is this side note here then before actually explaining it. The side note should perhaps be moved after the intro of opt-in RBF or before the segue, e.g. "Although Bitcoin Core may implement 'delayed RBF' at some point in the future, the prevalent…"

Reply:
@harding - I've converged to *emphasis* for introducing new terms. I think the document should be a consistent style, but please let me know if you think it can still be improved.

@Xekyo - I've rearranged the sentence as suggested. Is it clearer now?